Prevent any interactions while paragliding

Fuzss opened the pull request:
Hey there, really like the mod, nice job!

Just a little thing that bothered me: Even though both hands are holding the glider, you can still punch stuff and open containers while gliding. Also there is an inconsistence, where the arm swing shows in first person, but not in third person.

So I added a quick client side fix to disable all interactions while gliding. I also disabled drawing the highlight overlay, to make it a bit more clear that interactions are not possible while gliding.

Not sure where you want your client side event bus events, so I just put them in the main client class for now.

Fuzss committed:
prevent any interactions while paragliding

Tictim commented:
Hey, thanks for the PR.

Handling player interaction while paragliding has been defunct for years, and I appreciate you spending your time to fix it. But there's some things that I would like to discuss before merging.

- Paraglider exclusively uses `@EventBusSubscriber` annotation for event registration. The event handlers should probably go to `ParagliderClientEventHandler`.
- Measure should also be taken on server side in addition to the patch you provided; as the server can still process the action if client somehow bypasses the patch. The client shouldn't be trusted over such things.
  There is [half-broken attempt](https://github.com/Tictim/Paraglider/blob/release/1.18.2/src/main/java/tictim/paraglider/event/ParagliderEventHandler.java#L38) for server-side restriction from years ago. I have a suspicion that off hand checking part is the problematic part that renders the entire handler nonfunctional, because main hand actions are still going to pass. I'd appreciate it if you fix the code above as well.

Fuzss committed:
move events to dedicated events class

Fuzss committed:
cancel interactions server side too

Fuzss commented:
Great, I was looking for a dedicated events class, but didn't see it at first. Moved everything over there now.

I also added/fixed the requested server side checks. Seems good to go now from my testing.

Tictim committed:
Rearrange & cleanup

Tictim commented:
Thank you for the work! I will port the changes to other versions.
